source/core/GenericFunctions.cpp

```diff
@@ -762,7 +773,88 @@ VIREO_FUNCTION_SIGNATURET(VectorMaxMinOp, AggregateMaxAndMinInstruction)
 
     return _NextInstruction();
 }
+//------------------------------------------------------------
+// This function is used by the "Max and Min" primitive when first input is scalar and second input is array
+VIREO_FUNCTION_SIGNATURET(ScalarVectorMaxMinOp, AggregateMaxAndMinInstruction)
+{
+    TypedArrayCoreRef srcArrayY = _Param(VY);
+    TypedArrayCoreRef maxArray = _Param(VMax);
+    TypedArrayCoreRef minArray = _Param(VMin);
+    Instruction4<void, AQBlock1, AQBlock1, AQBlock1>* snippet = (Instruction4<void,
+        AQBlock1, AQBlock1, AQBlock1>*)_ParamMethod(Snippet());
+
+    IntIndex elementSizeY = srcArrayY->ElementType()->TopAQSize();
+    IntIndex elementSizeMax = maxArray->ElementType()->TopAQSize();
+    IntIndex elementSizeMin = minArray->ElementType()->TopAQSize();
+    IntIndex count = srcArrayY->Length();
+    // Resize output to size of input arrays
+    IntIndex* newDimensionLengths = srcArrayY->DimensionLengths();
+    IntIndex rank = srcArrayY->Rank();
+    maxArray->ResizeDimensions(rank, newDimensionLengths, true);
+    minArray->ResizeDimensions(rank, newDimensionLengths, true);
+
+    ArrayIterator srcArrayYIter(srcArrayY, rank, newDimensionLengths);
+    ArrayIterator destArrayMaxIter(maxArray, rank, newDimensionLengths);
+    ArrayIterator destArrayMinIter(minArray, rank, newDimensionLengths);
+
+    AQBlock1 *srcArrayYIterPtr = (AQBlock1 *)srcArrayYIter.Begin();
+    AQBlock1 *destArrayMaxIterPtr = (AQBlock1 *)destArrayMaxIter.Begin();
+    AQBlock1 *destArrayMinIterPtr = (AQBlock1 *)destArrayMinIter.Begin();
+    AQBlock1 *destArrayMaxEndIterPtr = destArrayMaxIterPtr + (count * elementSizeMax);
+    snippet->_p0 = _ParamPointer(SX);
+    while (destArrayMaxIterPtr < destArrayMaxEndIterPtr) {
+        snippet->_p1 = srcArrayYIterPtr;
+        snippet->_p2 = destArrayMaxIterPtr;
+        snippet->_p3 = destArrayMinIterPtr;
+        _PROGMEM_PTR(snippet, _function)(snippet);
+        srcArrayYIterPtr += elementSizeY;
+        destArrayMaxIterPtr += elementSizeMax;
+        destArrayMinIterPtr += elementSizeMin;
+    }
+
+    return _NextInstruction();
+}
+//------------------------------------------------------------
+// This function is used by the "Max and Min" primitive when first input is array and second input is scalar
+VIREO_FUNCTION_SIGNATURET(VectorScalarMaxMinOp, AggregateMaxAndMinInstruction)
```

Comment:
I wonder we should just support only ScalarVectorMaxMinOp and not support VectorScalarMaxMinOp at all in Vireo; and let LabVIEW swap the inputs while emitting VIA such that the first is always Scalar and the second is Vector when one of the inputs is scalar and other is vector. This will avoid unnecessary code duplication here and make the runtime smaller. I think it will be an acceptable limitation for VIA that the first input must be scalar ans the second one must be vector when one of the inputs is scalar and the other is vector. Probably, same can be done for ScalarVectorBinaryOp too. Thoughts?